akka-remote/src/main/scala/akka/remote/serialization/ArteryControlMessageSerializer.scala

```diff
@@ -91,39 +91,42 @@ private[akka] class ArteryControlMessageSerializer(val system: ExtendedActorSyst
   def deserializeQuarantined(quarantined: ArteryControlFormats.Quarantined): Quarantined =
     Quarantined(deserializeUniqueAddress(quarantined.getFrom), deserializeUniqueAddress(quarantined.getTo))
 
-  def serializeActorRef(ref: ActorRef): String =
-    if (ref == system.deadLetters) "DEADLETTER"
-    else Serialization.serializedActorPath(ref)
+  def serializeActorRef(ref: ActorRef): String = Serialization.serializedActorPath(ref)
+  def deserializeActorRef(str: String): ActorRef = system.provider.resolveActorRef(str)
 
-  def deserializeActorRef(str: String): ActorRef =
-    if (str == "DEADLETTER") system.deadLetters
-    else system.provider.resolveActorRef(str)
+  def serializeActorRefCompressionAdvertisement(adv: ActorRefCompressionAdvertisement): ArteryControlFormats.CompressionTableAdvertisement =
+    // FIXME: is it guaranteed that idx 0 is reserved for DeadLetters? In the best case, this knowledge should be managed in only one place.
+    serializeCompressionAdvertisement(adv)(serializeActorRef, _ != 0 /* 0 is reserved for DeadLetters and doesn't need to be serialized explicitly */ )
 
-  def serializeCompressionAdvertisement[T](adv: CompressionAdvertisement[T])(keySerializer: T ⇒ String): ArteryControlFormats.CompressionTableAdvertisement = {
+  def deserializeActorRefCompressionAdvertisement(bytes: Array[Byte]): ActorRefCompressionAdvertisement =
+    deserializeCompressionAdvertisement(bytes, deserializeActorRef, ActorRefCompressionAdvertisement, Seq(system.deadLetters → 0) /* add DeadLetters explicitly */ )
+
+  def serializeCompressionAdvertisement[T](adv: CompressionAdvertisement[T])(keySerializer: T ⇒ String, valueFilter: Int ⇒ Boolean = _ ⇒ true): ArteryControlFormats.CompressionTableAdvertisement = {
     val builder =
       ArteryControlFormats.CompressionTableAdvertisement.newBuilder
         .setFrom(serializeUniqueAddress(adv.from))
         .setTableVersion(adv.table.version)
 
     adv.table.map.foreach {
-      case (key, value) ⇒
+      case (key, value) if valueFilter(value) ⇒
```

Comment:
For DeadLetters it seems to be this:

https://github.com/akka/akka/blob/artery-dev/akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala#L120